### PR TITLE
Add the ability to turn a QR code into a hyperlink

### LINKED ISSUE #50: Hyperlink QR code image for HTML output
Hi again,

Would you consider a feature, or a pull request for a feature, to allow for hyperlinking the QR code when the output format is HTML? That way if someone is viewing the document but is unable/does not want to scan the QR code, they can click the image to access the same link? Basically wrapping the image tag in an anchor.

What do you think?

### PR DESCRIPTION
**Pull request by mattwarkentin**
Closes #50 

I basically only made changes in `drop_html()`. I pulled out the code for making the QR image into its own function, `make_qr_img()`, so that it was easier to add some conditional logic for whether to wrap it in an `<a>` tag as a hyperlink based on the new argument `qr_hyperlink`. I set `qr_hyperlink` to `FALSE` by default to be consistent with previous behaviour.

Let me know if you would like any changes!

### CODE REVIEW

**Comment by nucleic-acid** [comment]
Thanks for the PR, I'll have a closer look over the weekend!

To be honest, I think we could discuss whether QR codes should be links by default?
From the top of my head I can't think of situations where it breaks any functionality for existing applications. What do you think?

**Comment by mattwarkentin** [comment]
I agree, I am in favour of hyperlinking the QR codes, by default. I just tend to contribute PRs that avoid changing previous behaviour if at all possible, and leave it to the author/maintainer/you to decide on the final approach.

**Comment by mattwarkentin** [comment]
Just want to document this thought while it is in my head (we can spawn a new issue if desired). Is there a reason why the HTML snippets are stored in `visual_citations/` by default, rather than in a temporary directory that gets cleaned up when the R session is killed?

If my understanding is correct, the file(s) contained in `visual_citations/` are no longer necessary once `htmltools::includeHTML()` has been called to inject the HTML into the main file.

I understand storing the QR codes when `include_qr` is `link` or `link_svg`. But it seems to me that `visual_citations/` can live in a temporary directory instead of saving intermediate files into the users local directory. Thoughts?

**Review by nucleic-acid — APPROVED** [review]
Thanks for refactoring the code. It looks fine to be merged.

**Pull request merged**
